backend/src/test/java/com/empire/ArmyTest.java

```diff
@@ -1,280 +1,381 @@
+		assertEquals(80, a.raze(w, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
 		assertEquals(r.religion, Ideology.SWORD_OF_TRUTH);
 		assertEquals(0, r.constructions.size());
-		assertEquals(0, plainArmy.raze(world, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
+		assertEquals(0, a.raze(w, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
 
+		/*
+		//TODO: This test does not belong in the Army class, the Region should be responsible for determining how its religion has changed, this is impossible to properly/effectively test as written
 		// Razing temple changes religion.
 		r.constructions.add(Construction.makeTemple(Ideology.SWORD_OF_TRUTH, 100));
 		r.constructions.add(Construction.makeTemple(Ideology.SWORD_OF_TRUTH, 100));
 		r.constructions.add(Construction.makeTemple(Ideology.VESSEL_OF_FAITH, 100));
-		r.setReligion(null, world);
+		r.setReligion(null, w);
 		assertEquals(Ideology.SWORD_OF_TRUTH, r.religion);
-		assertEquals(80, plainArmy.raze(world, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
+		assertEquals(80, a.raze(w, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
 		assertEquals(Ideology.SWORD_OF_TRUTH, r.religion);
-		assertEquals(80, plainArmy.raze(world, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
+		assertEquals(80, a.raze(w, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
 		assertEquals(1, r.constructions.size());
 		assertEquals(Ideology.VESSEL_OF_FAITH, r.religion);
-		assertEquals(0, plainArmy.raze(world, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
+		assertEquals(0, a.raze(w, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
 		assertEquals(1, r.constructions.size());
+		*/
 
 		// Razing requires a minimum strength.
 		r.population = 1000000;
-		assertEquals(0, plainArmy.raze(world, "Raze temple Iruhan (Vessel of Faith)", null, 0, false), DELTA);
+		assertEquals(0, a.raze(w, "Raze temple Iruhan (Vessel of Faith)", null, 0, false), DELTA);
 
 		// Razing can raze multiple buildings in one action.
 		r.population = 100;
 		r.constructions.add(Construction.makeTemple(Ideology.SWORD_OF_TRUTH, 100));
 		r.constructions.add(Construction.makeTemple(Ideology.SWORD_OF_TRUTH, 100));
-		assertEquals(160, plainArmy.raze(world, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
+		assertEquals(160, a.raze(w, "Raze temple Iruhan (Sword of Truth)", null, 0, false), DELTA);
+	}
+
+	@Test
+	public void conquerBasic() {
+		a.kingdom = k2;
+		Region r = w.regions.get(0);
+		when(r.isLand()).thenReturn(true);
+		when(r.calcMinConquestStrength(w)).thenReturn(0.1);
+
+		HashSet<Region> conqueredRegions = new HashSet<>();
+		a.conquer(w, "Conquer", conqueredRegions, new HashMap<>(), new HashMap<>(), 0, new HashSet<>());
+		assertTrue(conqueredRegions.contains(r));
+		//assertEquals(k2, r.getKingdom());
```

Comment:
Hm, why not check this?

Reply:
Checked in a different way using Mockito's verification.